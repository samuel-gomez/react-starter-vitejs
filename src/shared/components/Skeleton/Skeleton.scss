.af-skeleton {
  display: inline-block;
  height: 47px;
  flex: 0 0 300px;
  background: linear-gradient(-45deg, #ecebf1 40%, #f2f1f5 50%, #ecebf1 60%);
  background-size: 400% 400%;
  animation: pulse 0.5s ease-in infinite;
  padding-left: 0;
  margin-left: 15px;
}

@keyframes pulse {
  0% {
    background-position: 0% 0%;
  }
  100% {
    background-position: -100% 0%;
  }
}
