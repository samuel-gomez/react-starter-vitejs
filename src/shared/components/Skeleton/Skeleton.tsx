import { withClassDefault, withClassModifier, WithClassModifierOptions, compose, identity } from '@axa-fr/react-toolkit-core';
import './Skeleton.scss';

type TSkeleton = {
  className?: string;
} & WithClassModifierOptions;

const DEFAULT_CLASSNAME = 'af-skeleton';

const Skeleton = ({ className }: TSkeleton) => <div aria-busy="true" aria-label="loader" role="status" className={className} />;

const enhance = compose(identity<TSkeleton>(), withClassDefault(DEFAULT_CLASSNAME), withClassModifier());

const Enhanced = enhance(Skeleton);
Enhanced.displayName = 'Skeleton';

export default Enhanced;
