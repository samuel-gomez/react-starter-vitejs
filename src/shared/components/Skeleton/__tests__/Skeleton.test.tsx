import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Skeleton from '../Skeleton';

describe('<Skeleton/>', () => {
  it('Render <Skeleton/> with default className', () => {
    const { asFragment, container } = render(<Skeleton />);
    expect(container.querySelector('.af-skeleton')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <Skeleton/> with other className', () => {
    const { asFragment, container } = render(<Skeleton className="other" />);
    expect(container.querySelector('.other')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <Skeleton/> with other className and custom modifier', () => {
    const { asFragment, container } = render(<Skeleton className="other" classModifier="custom" />);

    expect(container.querySelector('.other--custom')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should contain className .af-skeleton--loader when render Skelelon with "loader" classModifier', () => {
    const { asFragment, getByRole, container } = render(<Skeleton classModifier="loader" />);
    expect(getByRole('status')).toBeDefined();
    expect(container.querySelector('.af-skeleton--loader')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
