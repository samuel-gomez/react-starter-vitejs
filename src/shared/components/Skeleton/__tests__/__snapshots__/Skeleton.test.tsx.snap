// Vitest Snapshot v1

exports[`<Skeleton/> > Render <Skeleton/> with default className 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="loader"
    class="af-skeleton"
    role="status"
  />
</DocumentFragment>
`;

exports[`<Skeleton/> > Render <Skeleton/> with other className 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="loader"
    class="other"
    role="status"
  />
</DocumentFragment>
`;

exports[`<Skeleton/> > Render <Skeleton/> with other className and custom modifier 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="loader"
    class="other other--custom"
    role="status"
  />
</DocumentFragment>
`;

exports[`<Skeleton/> > Should contain className .af-skeleton--loader when render Skelelon with "loader" classModifier 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="loader"
    class="af-skeleton af-skeleton--loader"
    role="status"
  />
</DocumentFragment>
`;
